test/integration/policy_info_metric_test.go

```diff
@@ -0,0 +1,149 @@
+		By("Got the metrics route url: " + routeURL)
+	})
+	It("Checks that the endpoint does not expose metrics without auth", func() {
+		Eventually(func() interface{} {
+			resp, err := curlMetricsRoute()
+			if err != nil {
+				return err
+			}
+			return resp
+		}, defaultTimeoutSeconds, 1).Should(ContainSubstring("Unauthorized"))
+	})
+	It("Checks that endpoint has a HELP comment for the metric", func() {
+		token, err := oc("whoami", "-t")
+		Expect(err).To(BeNil())
+		Eventually(func() interface{} {
+			resp, err := curlMetricsRoute("--header", "Authorization: Bearer "+token)
+			if err != nil {
+				return err
+			}
+			return resp
+		}, defaultTimeoutSeconds, 1).Should(ContainSubstring("HELP " + metricName))
+	})
+	It("Checks that a compliant policy reports a metric of 0", func() {
+		By("Creating a compliant policy")
+		oc("apply", "-f", compliantPolicyYaml, "-n", userNamespace, "--kubeconfig="+kubeconfigHub)
+		Eventually(
+			getComplianceState(compliantPolicyName),
+			defaultTimeoutSeconds,
+			1,
+		).Should(Equal(policiesv1.Compliant))
+
+		By("Checking the policy metric")
+		token, err := oc("whoami", "-t")
+		Expect(err).To(BeNil())
+		regex := `(?m)` + metricName + `{.*policy="` + compliantPolicyName + `.*} 0$`
+		Eventually(func() interface{} {
+			resp, err := curlMetricsRoute("--header", "Authorization: Bearer "+token)
+			if err != nil {
+				return err
+			}
+			return resp
+		}, defaultTimeoutSeconds, 1).Should(MatchRegexp(regex))
+	})
+	It("Checks that a noncompliant policy reports a metric of 1", func() {
+		By("Creating a noncompliant policy")
+		oc("apply", "-f", noncompliantPolicyYaml, "-n", userNamespace, "--kubeconfig="+kubeconfigHub)
+		Eventually(
+			getComplianceState(noncompliantPolicyName),
+			defaultTimeoutSeconds,
+			1,
+		).Should(Equal(policiesv1.NonCompliant))
+
+		By("Checking the policy metric")
+		token, err := oc("whoami", "-t")
+		Expect(err).To(BeNil())
+		regex := `(?m)` + metricName + `{.*policy="` + noncompliantPolicyName + `.*} 1$`
+		Eventually(func() interface{} {
+			resp, err := curlMetricsRoute("--header", "Authorization: Bearer "+token)
+			if err != nil {
+				return err
+			}
+			return resp
+		}, defaultTimeoutSeconds, 1).Should(MatchRegexp(regex))
+	})
+	It("Cleans up", func() {
+		oc("delete", "-f", compliantPolicyYaml, "-n", userNamespace, "--kubeconfig="+kubeconfigHub)
+		oc("delete", "-f", noncompliantPolicyYaml, "-n", userNamespace, "--kubeconfig="+kubeconfigHub)
+		oc("delete", "route", "-n", ocmNS, "-l", propagatorMetricsSelector)
+	})
+})
+
+func oc(args ...string) (string, error) {
+	output, err := exec.Command("oc", args...).CombinedOutput()
+	if len(args) > 0 && args[0] != "whoami" {
+		fmt.Println(string(output))
+	}
+	return string(output), err
+}
+
+func curlMetricsRoute(args ...string) (string, error) {
+	argList := []string{"-k", "https://" + routeURL + "/metrics"}
+	argList = append(argList, args...)
+	output, err := exec.Command("curl", argList...).CombinedOutput()
```

Comment:
Could you please use [net/http](https://pkg.go.dev/net/http) instead? That's OS agnostic, you don't have to parse stdout/stderr, and it feels a lot cleaner.

Also, don't forget to add a timeout since we don't want the HTTP request to wait forever if that server is acting funky.

Reply:
That's a good idea